Gems/ROS2/Code/Include/ROS2/Sensor/Events/SensorEventSource.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+namespace ROS2
+{
+    struct SensorConfiguration;
+
+    //! Base class template for sensor event sources (based e.g. on TickBus or engine physics callback). Developer that wants to implement
+    //! new event source, should derive it from this class and signal m_sensorEvent based on chosen approach. Derived classes should rather
+    //! not auto connect any handler to m_sensorEvent, unless it is done to manage their internal logic. This event is for the final user
+    //! that can connect prepared event handler to it using ConnectToSourceEvent method.
```

Comment:
This class documentation is a bit unclear to me. Could you rephrase it?
It could be nice to clarify distinction and refer to EventSourceAdapter since typical use of sensors involves frequency.